src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java

```diff
@@ -47,6 +49,8 @@
 
   //Holdings specific Fields
   INSTANCE_ID("instanceId", "$.holdings.instanceId"),
+  HOLDING_NOTE_TYPE("holdingNoteTypeId", "$.holdings.notes[?(@.holdingsNoteTypeId=='{id}' && (!(@.staffOnly) || @.staffOnly == 'false'))].holdingsNoteTypeId", HOLDING_NOTE_TYPES),
+  HOLDING_NOTE_TYPE_STAFF_ONLY("holdingNoteTypeId.staffOnly", "$.holdings.notes[?(@.holdingsNoteTypeId=='{id}' && (!(@.staffOnly) || @.staffOnly == 'true'))].holdingsNoteTypeId", HOLDING_NOTE_TYPES),
```

Comment:
A record with no staff only will be matched by both these rules Igor. Please test an example against jsonpath that will give you a better idea.

Reply:
Agree, just remove ! sign before staffOnly in the second condition